src/process.cpp

```diff
@@ -18,248 +20,275 @@ void ManifoldPreprocess(Params &params, Model &m, ofstream &of)
+                    }
+                    ++rowIdx;
+                    ++top_row;
+                }
 
-        ++top_row;
-        rowIdx += p1;
-        for (size_t i = p1 + 1; i < (costSize + 1); ++i)
-        {
-          costMatrix[rowIdx] = costMatrix[top_row];
-          precostMatrix[rowIdx] = precostMatrix[top_row++];
-          rowIdx += i;
-          assert(rowIdx >= 0);
+                ++top_row;
+                rowIdx += p1;
+                for (size_t i = p1 + 1; i < (costSize + 1); ++i)
+                {
+                    costMatrix[rowIdx] = costMatrix[top_row];
+                    precostMatrix[rowIdx] = precostMatrix[top_row++];
+                    rowIdx += i;
+                    assert(rowIdx >= 0);
+                }
+            }
+            costMatrix.resize(erase_idx);
+            precostMatrix.resize(erase_idx);
         }
-      }
-      costMatrix.resize(erase_idx);
-      precostMatrix.resize(erase_idx);
     }
-  }
 
-  return h;
+    return h;
 }
 
-void Compute(ofstream &of, Model &mesh, Params &params)
+void Compute(ofstream& of, Model& mesh, Params& params)
 {
-  vector<Model> InputParts = {mesh};
-  vector<Model> parts, pmeshs;
-
-  clock_t start, end;
-  start = clock();
-  of << "#Points: " << mesh.points.size() << endl;
-  of << "#Triangles: " << mesh.triangles.size() << endl;
+    vector<Model> InputParts = { mesh };
+    vector<Model> parts, pmeshs;
+#ifdef _OPENMP
+    omp_lock_t writelock;
+    omp_init_lock(&writelock);
+#endif
+    clock_t start, end;
+    start = clock();
+    of << "#Points: " << mesh.points.size() << endl;
+    of << "#Triangles: " << mesh.triangles.size() << endl;
 
-  of << " - Decomposition (MCTS)" << endl;
-  cout << " - Decomposition (MCTS)" << endl;
+    of << " - Decomposition (MCTS)" << endl;
+    cout << " - Decomposition (MCTS)" << endl;
 
-  size_t iter = 0;
-  double cut_area;
-  while ((int)InputParts.size() > 0)
-  {
-    vector<Model> tmp;
-    of << "iter " << iter << " ---- "
-         << "waiting pool: " << InputParts.size() << endl;
-    cout << "iter " << iter << " ---- "
-         << "waiting pool: " << InputParts.size() << endl;
-    for (int p = 0; p < (int)InputParts.size(); p++)
+    size_t iter = 0;
+    double cut_area;
+    while ((int)InputParts.size() > 0)
     {
-      if (p % ((int)InputParts.size() / 10 + 1) == 0)
-        cout << "Processing [" << p * 100.0 / (int)InputParts.size() << "%]" << endl;
-
-      Model pmesh = InputParts[p], pCH;
-      Plane bestplane;
-      pmesh.ComputeVCH(pCH);
-      double h = ComputeHCost(pmesh, pCH, params.rv_k, params.resolution, params.seed, 0.0001, false);
+        vector<Model> tmp;
+        of << "iter " << iter << " ---- "
+            << "waiting pool: " << InputParts.size() << endl;
+        cout << "iter " << iter << " ---- "
+            << "waiting pool: " << InputParts.size() << endl;
+#ifdef _OPENMP
+#pragma omp parallel for default(none) shared(InputParts,params) private(cut_area)

```

Comment:
```suggestion
#pragma omp parallel for default(none) shared(InputParts,params,mesh,writelock,parts,pmeshs,tmp,cout) private(cut_area)
```
Make it work on Linux.